presto-main/src/main/java/com/facebook/presto/sql/planner/RowExpressionInterpreter.java

```diff
@@ -781,6 +798,42 @@ private SpecialCallResult tryHandleCast(CallExpression callExpression, List<Obje
                 return changed(null);
             }
 
+            // Simplify nested casts to varchar types.
+            if (optimizationLevel == OPTIMIZED && source instanceof CallExpression && isVarcharType(targetType)) {
+                CallExpression innerCall = (CallExpression) source;
+                Type innerType = innerCall.getType();
+                if (isVarcharType(innerType) && resolution.isCastFunction(innerCall.getFunctionHandle()) && innerCall.getArguments().size() == 1) {
+                    VarcharType narrowestType = narrowVarcharType((VarcharType) targetType, (VarcharType) innerType);
```

Comment:
@krk, we already are doing this checks in `TypeCoercion`. Can you use that instead since it is more generic?

Reply:
Feel free to ping me in slack if you have questions.